app/src/main/java/com/example/authapp/social/FacebookLoginProcedure.kt

```diff
@@ -0,0 +1,93 @@
+package com.example.authapp.social
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import android.view.View
+import android.widget.ProgressBar
+import android.widget.Toast
+import com.example.authapp.R
+import com.example.authapp.handler.PreferenceHandler
+import com.example.authapp.ui.UserDashboardActivity
+import com.facebook.AccessToken
+import com.facebook.CallbackManager
+import com.facebook.FacebookCallback
+import com.facebook.FacebookException
+import com.facebook.login.LoginManager
+import com.facebook.login.LoginResult
+import com.google.firebase.auth.FacebookAuthProvider
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseUser
+
+class FacebookLoginProcedure(private val context: Context) {
```

Comment:
Context should not be passed beyond the scope of the view. so please whatever context related task is happening here move it to the ui layer.